js/source/source_cache.js

```diff
@@ -349,7 +352,10 @@ SourceCache.prototype = util.inherit(Evented, {
                 if (this.findLoadedChildren(coord, maxCoveringZoom, retain)) {
                     retain[id] = true;
                 }
-                this.findLoadedParent(coord, minCoveringZoom, parentsForFading);
+                parentTile = this.findLoadedParent(coord, minCoveringZoom, parentsForFading);
+                if (parentTile) {
+                    this.addTile(parentTile.coord);
+                }
```

Comment:
Why did you remove `this.addTile` from `findLoadedParent` and then add it to all the call sites of `findLoadedParent`?